### PR TITLE
Feat/doc improvements

### LINKED ISSUE #268: Make it clear that .extraConfig and .logs.extraConfig cannot modify existing config, they only add

### LINKED ISSUE #269: Add generated doc that shows RBAC requirements
We've had requests for listing the full RBAC requirements for this Helm chart. We should be able to generate a doc that builds that.

### PR DESCRIPTION
**Pull request by petewall**
Fixes #269 
Fixes #268

### CODE REVIEW

File: charts/k8s-monitoring/Makefile [hunk 1]
```diff
@@ -21,3 +21,32 @@ ifdef HAS_HELM_DOCS
 else
 	docker run --rm --volume "$(shell pwd):/helm-docs" -u $(shell id -u) jnorwood/helm-docs:latest
 endif
+
+docs/RBAC.md:
+	echo "# RBAC Rules for the Kubernetes Monitoring Helm Chart" > docs/RBAC.md
+	echo "Several components have specialized RBAC rules to perform their work. This document will list the components, and their RBAC definitions." >> docs/RBAC.md
+	echo "Several components have specialized RBAC rules to perform their work. This document will list the components, and their RBAC definitions" >> docs/RBAC.md
```
Comment (skl) [inline charts/k8s-monitoring/Makefile:28]: Duplicate line?

```suggestion
```

File: charts/k8s-monitoring/docs/RBAC.md [hunk 2]
```diff
@@ -0,0 +1,446 @@
+# RBAC Rules for the Kubernetes Monitoring Helm Chart
+Several components have specialized RBAC rules to perform their work. This document will list the components, and their RBAC definitions.
+Several components have specialized RBAC rules to perform their work. This document will list the components, and their RBAC definitions
+## Grafana Agent
+```yaml
+- apiGroups:
+    - ''
```
Comment (skl) [inline charts/k8s-monitoring/docs/RBAC.md:7]: These empty entries look a bit weird

**Review by skl — APPROVED** [review]
Nice work!

File: charts/k8s-monitoring/docs/RBAC.md (hunk 2 above)
Comment (petewall) [reply charts/k8s-monitoring/docs/RBAC.md:7]: an empty api group means that it applies to the core API group, so for pods, services, nodes, etc...

**Pull request merged**